src/utils.js

```diff
@@ -74,40 +86,56 @@ export const int2hex = mNumber => {
  * @return {string}
  */
 export const num2hexstring = (num, size = 2, littleEndian = false) => {
+  if (typeof num !== 'number') throw new Error('num must be numeric')
+  if (num < 0) throw new RangeError('num is unsigned (>= 0)')
+  if (!Number.isSafeInteger(num)) throw new RangeError('num must be a safe integer')
   let hexstring = num.toString(16)
   hexstring = hexstring.length % size === 0 ? hexstring : ('0'.repeat(size) + hexstring).substring(hexstring.length)
   if (littleEndian) hexstring = reverseHex(hexstring)
   return hexstring
 }
 
 /**
- * Converts a number to a Fixed8 format string
+ * Converts a number to a Fixed8 format hex string
  * @param {number} num
+ * @param {number} size output size in hex chars
  * @return {string} number in Fixed8 representation.
  */
-export const num2fixed8 = (num) => {
-  const hexValue = Math.round(num * 100000000).toString(16)
-  return reverseHex(('0000000000000000' + hexValue).substring(hexValue.length))
+export const num2fixed8hex = (num, size = 16) => {
```

Comment:
Fixed8 is a string format that represents a number with built in accuracy of 8 decimals places. The length of the string is dependent on the number of of trailing zeroes but it can always be represented as a length of 16 characters (8 bytes).

Is there any case where you will enforce a length that is not 16? I do think that there is no need for `fixed8hex` though because there isn't a different representation used in the system except by hexstring

Reply:
`fixed8` to me refers to the base 10 form and not the hex form, this is just what I gathered through digging through the neo sources and my initial impressions upon reading this code. we can roll that back if you disagree - it's not a big deal.

I can't think of a situation where we would specify a length other than 16 in neon-js but some contracts may be able to use it. For example, my SC puts a GAS amount in 5 bytes for a maximum of ~5497.5 GAS - (2^40)/2. This is for optimizing storage space if it is already known that it won't exceed that number.

Reply:
On this note, it would be good to open up discussion about whether we should always specify `size` in bytes and avoid confusion of hex/byte lengths

Reply:
I do know the NEO nodes return shaved fixed8 (fixed8s with leading zeroes shaved off). 

`Fixed8` to me was always an unreadable encoded form so hex for me. If i wanted base10, I will be using `number`? On a side note, Im not sure if `number` can handle the maximum value of `Fixed8`.

Flexibility is welcomed  for the method so i am ok. It also doesnt break existing implementations.

`size` wise, I think I will lean towards byte lengths so that its clear coming from C#. Hexstrings is just chosen to be the main representation of data used in this repo.

Reply:
Alright, I could make the changes to use byte sizes in the utils methods.

Reply:
Will do that as a separate PR. ^